rs/tests/crypto/ingress_verification_test.rs

```diff
@@ -596,24 +621,36 @@ pub fn requests_with_delegations_with_targets(env: TestEnv) {
                     )
                     .await;
 
-                    if scenario.expect_success {
+                    if scenario.expect_success() {
                         assert_eq!(
-                            read_state_result, 200,
+                            response.status(),
+                            200,
                             "Test scenario {} (read_state) using {api_ver} unexpectedly failed",
                             scenario.note
                         );
                     } else {
                         // Which error code is returned depends on API version and the specific scenario
                         assert!(
-                            read_state_result == 400 || read_state_result == 403,
+                            response.status() == 400 || response.status() == 403,
                             "Test scenario {} (read_state) using {api_ver} unexpectedly succeeded",
                             scenario.note
                         );
+
+                        let err_msg = scenario.expected_err.clone().unwrap();
+
+                        // The read request sometimes returns a different error message
+                        // than update/query depending on the type of malformed req.
+
+                        if err_msg.contains("expected at most") {
+                            response.expect_text_error(&err_msg);
+                        } else {
+                            response.expect_text_error("The user tries to access request IDs for canisters not belonging to sender delegation targets.");
+                        }
```

Comment:
As the spec doesn't say anything beyond "4xx HTTP status" for errors, we could even omit asserting the actual message. I'm also fine with keeping the assertion.